File: app/src/main/java/com/example/datacollector/presentation/MainActivity.java
package com.example.datacollector.presentation;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.ComponentActivity;
import androidx.annotation.Nullable;

import com.example.datacollector.R;
import com.example.datacollector.presentation.MoveSense.MovesenseActivity;
import com.example.datacollector.presentation.MoveSense.bluetooth.BluetoothStatusMonitor;
import com.example.datacollector.presentation.MoveSense.bluetooth.MdsRx;
import com.example.datacollector.presentation.MoveSense.bluetooth.RxBle;
import com.example.datacollector.presentation.location.LocationActivity;
import com.example.datacollector.presentation.heartrate.HeartRateActivity;


public class MainActivity extends ComponentActivity implements View.OnClickListener {

    private final String TAG = this.getClass().getName() + "UserDebug";


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTheme(android.R.style.Theme_DeviceDefault);
        setContentView(R.layout.activity_main);
        Button movesenseButton = findViewById(R.id.movesense_start_button);
        Button heartRateButton = findViewById(R.id.heart_rate_start_button);
        Button locationButton = findViewById(R.id.location_start_button);
        movesenseButton.setOnClickListener(this);
        heartRateButton.setOnClickListener(this);
        locationButton.setOnClickListener(this);

        initialServices();

        startActivity(new Intent(this, MovesenseActivity.class));

    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        if (id == R.id.movesense_start_button) {
            startActivity(new Intent(this, MovesenseActivity.class));
        } else if (id == R.id.heart_rate_start_button) {
            startActivity(new Intent(this, HeartRateActivity.class));
        } else if (id == R.id.location_start_button) {
            startActivity(new Intent(this, LocationActivity.class));
        }
    }


    public void initialServices(){
        // Initialize RxBleWrapper
        RxBle.Instance.initialize(this);
        // Initialize MDS
        MdsRx.Instance.initialize(this);
        BluetoothStatusMonitor.INSTANCE.initBluetoothStatus();
    }

}

File: app/src/main/java/com/example/datacollector/presentation/MainActivity.kt
/* While this template provides a good starting point for using Wear Compose, you can always
 * take a look at https://github.com/android/wear-os-samples/tree/main/ComposeStarter to find the
 * most up to date changes to the libraries and their usages.
 */

package com.example.datacollector.presentation

import android.os.Bundle
import android.util.Log
import androidx.activity.ComponentActivity
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import androidx.health.services.client.MeasureCallback
import androidx.health.services.client.MeasureClient
import androidx.health.services.client.data.Availability
import androidx.health.services.client.data.DataPointContainer
import androidx.health.services.client.data.DataType
import androidx.health.services.client.data.DataTypeAvailability
import androidx.health.services.client.data.DeltaDataType
import kotlinx.coroutines.flow.MutableStateFlow
import android.content.Intent
import android.widget.Button
import com.example.datacollector.R
import com.example.datacollector.presentation.Bluetooth.BLEScanActivity

class MainActivity : ComponentActivity() {

    private lateinit var measureClient: MeasureClient

    // Store HeartRateFlow data
    private val heartRateFlow = MutableStateFlow<Double?>(null)

    // Callback function
    val heartRateCallback = object : MeasureCallback {

        override fun onAvailabilityChanged(
            dataType: DeltaDataType<*, *>,
            availability: Availability
        ) {
            if (availability is DataTypeAvailability) {
                Log.d("HeartRate", "Availability changed: $availability")
            }
        }

        override fun onDataReceived(data: DataPointContainer) {
            // Inspect data points.
            val heartRateData = data.getData(DataType.HEART_RATE_BPM)
            if (heartRateData.isNotEmpty()) {
                val latestHeartRate = heartRateData.lastOrNull()?.value ?: return
                heartRateFlow.value = latestHeartRate
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        installSplashScreen()
        super.onCreate(savedInstanceState)
        setTheme(android.R.style.Theme_DeviceDefault)
        Log.i("MainActivityUserDebug", "MainActivity onCreate")

        setContentView(R.layout.main_activity)


        val bleScanButton = findViewById<Button>(R.id.bleScanButton)
        val heartrateButton = findViewById<Button>(R.id.heartrateButton)


        bleScanButton.setOnClickListener {
            Log.i("MainActivityUserDebug", "BLE Button Clicked")
            val intent = Intent(this, BLEScanActivity::class.java)
            startActivity(intent)
        }

        heartrateButton.setOnClickListener {
            Log.i("MainActivityUserDebug", "HeartRate Button Clicked")
            val intent = Intent(this, HeartRateActivity::class.java)
            startActivity(intent)
        }


    }

    override onC
}



